core/server/data/migrations/versions/4.3/08-migrate-members-signup-setting.js

```diff
@@ -0,0 +1,96 @@
+        if (!oldSetting) {
+            logging.warn('Could not find setting `members_allow_free_signup`, skipping update of `members_signup_access` setting');
+            return;
+        }
+
+        const newSetting = await options.transacting('settings')
+            .where('key', 'members_signup_access')
+            .select('value')
+            .first();
+
+        const migrateValue = oldSetting.value ? 'all' : 'invite';
+
+        if (newSetting) {
+            // new setting already exists, *update* with migrated value
+            logging.info('Updating `members_signup_access` setting with value from `members_allow_free_signup`');
+            await options.transacting('settings')
+                .where('key', 'members_signup_access')
+                .update('value', migrateValue);
+        } else {
+            // setting does not exist yet, *create* with migrated value
+            logging.info('Creating `members_signup_access` setting with value from `members_allow_free_signup`');
+
+            const now = new Date();
+
+            const newSettingData = {
+                id: ObjectId.generate(),
+                group: 'members',
+                key: 'members_signup_access',
+                value: migrateValue,
+                type: 'string',
+                created_at: now,
+                updated_at: now,
+                created_by: oldSetting.created_by,
+                updated_by: oldSetting.updated_by
+            };
+
+            await options.transacting('settings').insert(newSettingData);
+        }
+
+        logging.info('Deleting `members_allow_free_signup` setting');
+        await options.transacting('settings')
+            .where('key', 'members_allow_free_signup')
+            .del();
+    },
+
+    async down(options) {
+        const newSetting = await options.transacting('settings')
+            .where('key', 'members_signup_access')
+            .select('value')
+            .first();
+
+        const oldSetting = await options.transacting('settings')
+            .where('key', 'members_allow_free_signup')
+            .select('value')
+            .first();
+
+        if (!newSetting && oldSetting) {
+            logging.warn('Could not find setting `members_signup_access`, skipping rollback of `members_allow_free_signup` setting');
+            return;
+        }
+
+        if (!newSetting && !oldSetting) {
+            logging.warn('No members signup access settings found, skipping rollback');
+            return;
+        }
+
+        // this can potentially be lossy if going from "nobody" but it matches to the
+        // most appropriate setting available in earlier versions of Ghost
+        const rollbackValue = newSetting.value === 'all' ? true : false;
+
+        logging.info('Creating `members_allow_free_signup` based on value from `members_signup_access`');
+        await options.transacting('settings')
+            .where('key', 'members_signup_access')
```

Comment:
I believe this should be `members_allow_free_signup` as per the comment

Reply:
Good catch, this wasn't working as intended. Updated to insert/update based on presence of the old key

Reply:
Shouldn't the new version still be changed to `members_allow_free_signup`?

Reply:
🤦 too much copy/paste there. Fixed